youtube_dl/extractor/bbc.py

```diff
@@ -1074,6 +1086,85 @@ def _real_extract(self, url):
                     'subtitles': subtitles,
                 }
 
+        # simorgh-based playlist (see https://github.com/bbc/simorgh)
+        # JSON assigned to window.SIMORGH_DATA in a <script> element
+        simorgh_data = self._parse_json(
+            self._search_regex(
+                r'window\.SIMORGH_DATA\s*=\s*(\{[^<]+})\s*</',
+                webpage, 'simorgh playlist', default='{}'),
+            playlist_id, fatal=False)
+        # legacy media, video in promo object (eg, http://www.bbc.com/mundo/video_fotos/2015/06/150619_video_honduras_militares_hospitales_corrupcion_aw)
+        playlist = try_get(simorgh_data, lambda x: x['pageData']['promo']['media']['playlist']) or []
+        if playlist:
+            media = simorgh_data['pageData']['promo']
+            if media['media'].get('format') == 'video':
+                media.update(media['media'])
+                formats = []
+                keys = {'url', 'format', 'format_id', 'language', 'quality', 'tbr', 'resolution'}
+                for format in playlist:
+                    if not (format.get('url') and format.get('format')):
+                        continue
+                    bitrate = format.pop('bitrate')
+                    if bitrate:
+                        bitrate = re.sub(r'000\s*$', 'kbps', bitrate)
+                        format['tbr'] = parse_bitrate(bitrate)
+                    format['language'] = media.get('language')
+                    # format id: penultimate item from the url split on _ and .
+                    (fmt,) = re.split('[_.]', format['url'])[-2:][:1]
+                    format['format_id'] = '%s_%s' % (format['format'], fmt)
+                    if not format.get('resolution'):
+                        format['resolution'] = fmt
```

Comment:
Resolution is used for textual representation of width and height and typically automatically deduced from these meta fields.

Reply:
In this case the playlist data from the embedded JSON (unlike BBC `programme` playlists) doesn't appear to include video width or height, so these could only be found by parsing a selected media item. The test page has aspect ratio (global) and bitrate for each video. To populate the `resolution` column of `-F` output with the best information available, I try for a `resolution` (just in case) and otherwise use the resolution string `fmt` extracted as above. It might be more realistic to try for width/height instead, which I've implemented in the new version.